changes/804.doc.md

```diff
@@ -0,0 +1,3 @@
+Add the add_component method to `api.special_endpoints.ActionRowBuilder`
+Add `nox` and `towncrier` requirements to `CONTRIBUTING.md`
+Add docstrings to the remaining undocumented `GatewayBot` methods
```

Comment:
Sorry to say, but all these were already documented. They inherited the docstring from `GatewayBotAware`. Maybe a comment to specify this would be better?